doc/cabal-project.rst

```diff
@@ -703,6 +703,10 @@ feature was added.
 
     The command line variant of this flag is ``--compiler=ghc``.
 
+    At the moment, it's not possible to set :cfg-field:`compiler` on a
+    per-package basis, but eventually we plan on relaxing this
+    restriction. If this is something you need, give us a shout.
```

Comment:
Is there a plan? Can we promise that? I'd rather remove this promise (also from `with-compiler` field), as I simply don't see how it would be possible (without larger `cabal.project` overhaul).

Reply:
I don't know if there is a plan. But it would be a nice thing to have given the multitude of web haskell compilers. I can remove the promise if you would like?

Reply:
That line was introduced 5 years ago in https://github.com/haskell/cabal/commit/99d48fa0fe930e019ff940d59c152069d1a2aaf8

In that time per-package compiler setting didn't become any closer to be true. I would say that the best which could happen is that we get `if (..)` pragmas to `cabal.project` so same `cabal.project` file could be reused for different compilers (versions or kinds).

But having the same `cabal build all` building with two different compilers doesn't seem likely.
Thus, jremove `, but we plan...` - we don't. (In general, there should be links to open issues if there are plans)